### ISSUE TITLE
[API Proposal]: Record Default Constructor Accessibility

### ISSUE THREAD

**Issue by Foxtrek64** [dotnet/runtime#72702]
### Background and motivation

Currently with record types, whether record classes or record structs, the default constructor is always public. There are times where non-public constructors are relevant, however to do so, users must implement the properties and constructor manually.

There are certainly cases where this is better, but it removes the utility of the default constructor for record types.

```cs
// old
public sealed record class Person(string Name, int Age);

// new
public sealed record class Person
{
    public string Name { get; }
    public int Age { get; }
    
    private Person(string name, int age)
    {
        Name = name;
        Age = age;
     }
 }
 ```

### API Proposal

I would like to propose an additional syntax for record types, as follows:

```csharp
public sealed record class Person private(string Name, int Age);
public sealed record struct Person protected internal(string Name, int Age);
```
This would compile into a class or struct as normal, but the specified accessibility modifier would be applied to the generated constructor.

Necessarily, the compiler will need instructions that if an accessibility modifier is not applied, then it should be assumed to be `public` in order to maintain backwards compatibility with existing record types.

Optionally, an analyzer could also be created to recommend removing the `public` modifier for simplification reasons, 

### API Usage

The proposal and the usage are pretty much the same thing.

### Alternative Designs

The placement of the private modifier is based off of Kotlin's class specification. Here is an example of theirs:
```kt
class Foo private constructor(Bar bar) { }
```
That said, we are not beholden to any design because of another language's decisions, so if a different placement is desired for different reasons, I take no issue with this.

### Risks

So long as we make the modifier optional and specify that no modifier is the same as public, there are no risks to this change.
